wire-ios-sync-engine/Source/Services/SupportedProtocolsService.swift

```diff
@@ -31,29 +31,48 @@ final class SupportedProtocolsService: SupportedProtocolsServiceInterface {
     private let featureRepository: FeatureRepositoryInterface
     private let userRepository: UserRepositoryInterface
     private let logger = WireLogger(tag: "supported-protocols")
+    private let context: NSManagedObjectContext
+    let oneOnOneResolver: OneOnOneResolverInterface
 
     // MARK: - Life cycle
 
-    public convenience init(context: NSManagedObjectContext) {
+    public convenience init(
+        context: NSManagedObjectContext,
+        oneOnOneResolver: OneOnOneResolverInterface
+    ) {
         self.init(
             featureRepository: FeatureRepository(context: context),
-            userRepository: UserRepository(context: context)
+            userRepository: UserRepository(context: context),
+            oneOnOneResolver: oneOnOneResolver,
+            context: context
         )
     }
 
     init(
         featureRepository: FeatureRepositoryInterface,
-        userRepository: UserRepositoryInterface
+        userRepository: UserRepositoryInterface,
+        oneOnOneResolver: OneOnOneResolverInterface,
+        context: NSManagedObjectContext
     ) {
         self.featureRepository = featureRepository
         self.userRepository = userRepository
+        self.oneOnOneResolver = oneOnOneResolver
+        self.context = context
     }
 
     // MARK: - Methods
 
     func updateSupportedProtocols() {
         let selfUser = userRepository.selfUser()
+        let previousProtocols = selfUser.supportedProtocols
+
         selfUser.supportedProtocols = calculateSupportedProtocols()
+
+        if previousProtocols != selfUser.supportedProtocols {
+            Task {
+                try await oneOnOneResolver.resolveAllOneOnOneConversations(in: context)
+            }
+        }
```

Comment:
By resolving the one on one conversations here we're adding a new hidden responsibility of the `SupportedProtocolService`. In my opinion, this class is responsible only for calculating the current supported protocols of the self user and storing it on the self user.

I suggest to create a delegate method so that we can report externally that the supported protocols of the self user changed, something like:

```swift
delegate?.supportedProtocolsService(self, didCaluculateNewSupportedProtocols: selfUser.supportedProtocols)
```

Reply:
I agree this is not a good idea and turns around responsibilities as it was implemented by the slow sync for example right now... ❌

Reply:
I started with that approach initially but after talking with @netbe he suggested to go another route. @johnxnguyen Using the delegate it will make a bit tricky to pass the resolver up to many classes. I remember having this problem as I initially implemented it that way. Thoughts?